src/Drips.sol

```diff
@@ -107,10 +107,10 @@ library Drips {
         uint256 userId,
         uint256 assetId
     ) internal view returns (uint32 cycles) {
-        uint32 collectedCycle = s.dripsStates[assetId][userId].nextCollectedCycle;
-        if (collectedCycle == 0) return 0;
+        uint32 nextCollectedCycle = s.dripsStates[assetId][userId].nextCollectedCycle;
         uint32 currFinishedCycle = _currTimestamp() / cycleSecs;
-        return currFinishedCycle + 1 - collectedCycle;
+        if (nextCollectedCycle == 0 || nextCollectedCycle > currFinishedCycle) return 0;
+        return currFinishedCycle - nextCollectedCycle + 1;
```

Comment:
Okay, this makes sense.

Basically, before we had the situation. That `nextCollectedCycle`  could be in future and this method would have resulted in an underflow by substracting `currFinishedCycle` with `collectedCycle`?

Reply:
No, that was safe, `collectedCycle` was being set to the value equal to `_currTimestamp() / cycleSecs + 1` when initializing it in `setDrips`, so this subtraction could never underflow. That was inefficient but sound.